server/src/main/java/io/crate/planner/operators/JoinPlanBuilder.java

```diff
@@ -72,26 +69,8 @@ static LogicalPlan buildJoinTree(List<AnalyzedRelation> from,
         }
         Map<Set<RelationName>, Symbol> queryParts = QuerySplitter.split(whereClause);
         List<JoinPair> allJoinPairs = convertImplicitJoinConditionsToJoinPairs(joinPairs, queryParts);
-        boolean optimizeOrder = true;
-        for (var joinPair : allJoinPairs) {
-            if (hasAdditionalDependencies(joinPair)) {
-                optimizeOrder = false;
-                break;
-            }
-        }
         LinkedHashMap<Set<RelationName>, JoinPair> joinPairsByRelations = buildRelationsToJoinPairsMap(allJoinPairs);
-        Iterator<RelationName> it;
-        if (optimizeOrder) {
-            Collection<RelationName> orderedRelationNames = JoinOrdering.getOrderedRelationNames(
-                Lists2.map(from, AnalyzedRelation::relationName),
-                joinPairsByRelations.keySet(),
-                queryParts.keySet()
-            );
-            it = orderedRelationNames.iterator();
-        } else {
-            it = Lists2.map(from, AnalyzedRelation::relationName).iterator();
-        }
-
+        Iterator<RelationName> it = Lists2.map(from, AnalyzedRelation::relationName).iterator();
```

Comment:
```suggestion
        Iterator<RelationName> it = Lists2.mapLazy(from, AnalyzedRelation::relationName).iterator();
```

No need to materialize the full list in memory if we're only using its iterator